### PR TITLE
[Application] add step learning rate scheduler example

### PR DESCRIPTION
**Pull request by lhs8928**
- Added how to use step learning rate scheduler in mnist application

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: Applications/MNIST/jni/main.cpp [hunk 1]
```diff
@@ -270,6 +271,18 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
+  try {
+    auto optimizer = ml::train::createOptimizer("adam");
+    auto lr_scheduler = ml::train::createLearningRateScheduler("step");
+    lr_scheduler->setProperty({"learning_rate=0.0001, 0.00005"});
+    lr_scheduler->setProperty({"iteration=10"});
```
Comment (jijoongmoon) [inline Applications/MNIST/jni/main.cpp:278]: It might be more clear how to use the step learning rate scheduler if more learning rate table is added.
```suggestion
    lr_scheduler->setProperty({"learning_rate=0.0001, 0.00009, 0.00007, 0.00005"});
    lr_scheduler->setProperty({"iteration=4, 6, 15"});
```

Comment (lhs8928) [reply Applications/MNIST/jni/main.cpp]: fixed :)

**Review by songgot — APPROVED** [review]
Looks good to me

File: Applications/MNIST/jni/main.cpp [hunk 2]
```diff
@@ -270,6 +271,19 @@ int main(int argc, char *argv[]) {
...
 
+  try {
+    auto optimizer = ml::train::createOptimizer("adam");
+    auto lr_scheduler = ml::train::createLearningRateScheduler("step");
+    lr_scheduler->setProperty(
+      {"learning_rate=0.0001, 0.00009, 0.00007, 0.00005"});
+    lr_scheduler->setProperty({"iteration=4, 6, 15"});
+    optimizer->setLearningRateScheduler(std::move(lr_scheduler));
+    model->setOptimizer(std::move(optimizer));
+  } catch (std::exception &e) {
```
Comment (jihochu) [inline Applications/MNIST/jni/main.cpp:282]:
```suggestion
  } catch (const std::exception &e) {
```
How about adding const ?

File: Applications/MNIST/jni/main.cpp [hunk 3]
```diff
@@ -270,6 +271,19 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
+  try {
+    auto optimizer = ml::train::createOptimizer("adam");
+    auto lr_scheduler = ml::train::createLearningRateScheduler("step");
+    lr_scheduler->setProperty(
```
Comment (jihochu) [inline Applications/MNIST/jni/main.cpp:277]: How about adding empty line after declare variable?

Comment (lhs8928) [reply Applications/MNIST/jni/main.cpp:278]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: Applications/MNIST/jni/main.cpp [hunk 4]
```diff
@@ -197,8 +198,8 @@ int getSample(float **outVec, float **outLabel, bool *last, void *user_data) {
 
 #if defined(APP_VALIDATE)
 TEST(MNIST_training, verify_accuracy) {
-  EXPECT_FLOAT_EQ(training_loss, 2.5698349);
-  EXPECT_FLOAT_EQ(validation_loss, 2.5551746);
+  EXPECT_FLOAT_EQ(training_loss, 2.586082);
+  EXPECT_FLOAT_EQ(validation_loss, 2.5753405);
```
Comment (jijoongmoon) [inline Applications/MNIST/jni/main.cpp:202]: This is because of the different learning rate scheduling.

**Review by jijoongmoon — APPROVED** [review]
LGTM!
